decode_response.go

```diff
@@ -31,9 +32,27 @@ func (sp *SAMLServiceProvider) ValidateEncodedResponse(encodedResponse string) (
 		return nil, err
 	}
 
-	response, err := sp.validationContext().Validate(doc.Root())
-	if err != nil && !sp.SkipSignatureValidation || response == nil {
-		return nil, err
+	response := doc.Root()
+	if !sp.SkipSignatureValidation {
+		response, err = sp.validationContext().Validate(doc.Root())
+		if err == errors.New("Missing signature referencing the top-level element") {
```

Comment:
I just pulled out that error into a variable in the other library, can you update this to reference `ErrMissingSignature`?

https://github.com/russellhaering/goxmldsig/commit/ea347e26e03d8ba6c60f1be76cfd59ba7d5e0a42